packages/vite/src/node/utils.ts

```diff
@@ -76,6 +76,7 @@ export function unwrapId(id: string): string {
 
 export const flattenId = (id: string): string =>
   id
+    .replace(/\/#/g, '/[sharp]')
     .replace(/[/:]/g, '_')
     .replace(/\./g, '__')
     .replace(/(\s*>\s*)/g, '___')
```

Comment:
I think we should keep the current scheme, even if it is starting to break (if we have to add more, we could start using `__x__` with different letters for each).

And we could support a sharp anywhere in the file name, no? We don't need to restrict it to sharps after a slash.
```suggestion
    .replace(/[/:]/g, '_')
    .replace(/\./g, '__')
    .replace(/(\s*>\s*)/g, '___')
    .replace(/#/g, '____')
```

Reply:
Good point. L82 `'____'` seems to have one more `'_'` than L81's `'___'`. Is it intended or a misspelling?

Reply:
Ah, wait, L81 is for nested imports `a > b` 🤔, that ends as `a___b`. Still, it is safer that `a > b.js` doesn't end up with the same path as `a#b.js`

Reply:
Well, it seems that there is no difference between `a___.js` and `a____.js` at first glance, especially in a complex file path.

Reply:
That isn't a problem, the idea is just to have unique ids. But we may want to review the current scheme because `::` and `.` ends up being mapped to the same value.
Maybe we should use something like this in the future:
```
. -> __d__
/ -> __s__
: -> __c__
> -> __n__
# -> __h__
```
I think it is fine for the current PR to continue the scheme as a way to fix the issue, and then we could do a new one that reworks `flattenId` in a more holistic way.

Reply:
Done